PsfShimMonitor/KernelTrace.cs

```diff
@@ -0,0 +1,1356 @@
+﻿//-------------------------------------------------------------------------------------------------------
+// Copyright (C) TMurgent Technologies. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+//
+// NOTE: PsfMonitor is a "procmon"-like display of events captured via the PSF TraceShim.
+
+using Microsoft.Diagnostics.Tracing;  // consumer
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Session; // controller
+using System;
+using System.ComponentModel;  // backgroundworker
+using System.Threading;
+using System.Windows;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;  // ObservableCollection
+
+namespace PsfMonitor
+{
+    public partial class MainWindow : Window
+    {
+        public TraceEventSession myTraceEventSession_ProcsKernel = null;
+        public bool rememberToDisableSession_ProcsKernel = false;
+        public bool PleaseStopCollecting = false;
+        public int FilterOnProcessId = -1;
+
+        public List<EventItem> _TKernelEventListItems = new List<EventItem>();
+        public Object _TKernelEventListsLock = new Object();
+
+        Dictionary<UInt64, string> _KCBs = new Dictionary<UInt64, string>();
+        Dictionary<UInt64, string> _TKCBs = new Dictionary<UInt64, string>();
+        public Object _TKCBsListLock = new object();
+
+        private int MAX_KCBS = 100000;
+
+        private void KernelTraceInBackground_Start()
+        {
+            if (kerneleventbgw == null)
```

Comment:
Is there a reason you are checking for null then IsBusy?

Reply:
Standard pattern from Microsoft samples for background workers.  You don't assume caller pre-created the worker and create it if needed.  Then you check to ensure there wasn't one already running.

Reply:
I see.  This method only gets called once though.  From MainWindow in MainWindow.Zaml.cs.  Do you need it here, or is this a guard in case someone else calls this method a second time?

Reply:
It is a guard.  Background functions and delegates are easy for someone to mess up if they aren't used to it.